nft-minter/src/nft_minting.rs

```diff
@@ -126,6 +128,51 @@ pub trait NftMintingModule:
         self.nft_giveaway_event(&brand_id, &tier, total);
     }
 
+    #[payable("*")]
+    #[endpoint(nftUpgrade)]
+    fn nft_upgrade(&self, brand_id: BrandId<Self::Api>, tier: TierName<Self::Api>) {
+        let old_nft = self.call_value().single_esdt();
+        let nft_token_id = self.nft_token(&brand_id).get_token_id();
+        let nft_amount = BigUint::from(NFT_AMOUNT);
+        require!(
+            old_nft.token_identifier == nft_token_id && old_nft.amount == nft_amount,
+            "Invalid payment"
+        );
+
+        let nft_id = self.get_next_random_id(&brand_id, &tier);
+        let brand_info: BrandInfo<Self::Api> = self.brand_info(&brand_id).get();
+        let nft_name = self.get_nft_name_with_tag(brand_info.token_display_name, nft_id);
+
+        let sc_address = self.blockchain().get_sc_address();
+        let old_nft_data = self.blockchain().get_esdt_token_data(
+            &sc_address,
+            &old_nft.token_identifier,
+            old_nft.token_nonce,
+        );
+
+        require!(
+            old_nft_data.royalties > ROYALTIES_THRESHOLD,
```

Comment:
Why this `require`? You only allow upgrade for NFTs that have royalties higher than 10%?